pkg/orchestrator/pod.go

```diff
@@ -127,9 +127,25 @@ func ScrubContainer(c *v1.Container, scrubber *DataScrubber) {
 		}
 	}()
 
-	// scrub command line
-	scrubbedCmd, _ := scrubber.ScrubSimpleCommand(c.Command)
-	c.Command = scrubbedCmd
+	// scrub args and commands
+	merged := append(c.Command, c.Args...)
+	words := 0
+	for _, cmd := range c.Command {
+		words += len(strings.Split(cmd, " "))
+	}
+
+	scrubbedMergedCommand, changed := scrubber.ScrubSimpleCommand(merged) // return value is split if has been changed
+	if !changed {
+		return // no change has happened, no need to go further down the line
+	}
+
+	// if part of the merged command got scrubbed the updated value will be split, even for e.g. c.Args only if the c.Command got scrubbed
+	if len(c.Command) > 0 {
+		c.Command = scrubbedMergedCommand[:words]
+	}
+	if len(c.Args) > 0 {
+		c.Args = scrubbedMergedCommand[words:]
+	}
```

Comment:
Could a custom regex / sensitive pattern "replace two words into one"  one day and break this, or would this not be possible? (looks like we don't allow custom regexps to be set for the orchestrator part atm, but just to confirm).

Reply:
you are right, currently we do not support custom regex.

The regex pattern we will allow in the future will only be prechecked here: https://github.com/DataDog/datadog-agent/blob/b2637a18ba65316b5620c4023af53fc34c14cbce/pkg/process/config/data_scrubber.go#L56 

Which basically only accepts `([a-zA-z0-9_]) or wildcards *`. So the regex can only do basic checks and not grouping and replacing. 

We could add future tests for this in case we introduce the regex matching again.